### ISSUE TITLE
docker build issues since 1.0.0

### ISSUE THREAD

**Issue by aviaviavi** [aviaviavi/toodles#54]
it appears docker stopped working with the latest release

```
Step 9/16 : RUN stack install --ghc-options '-optl-static -fPIC -optc-Os'
 ---> Running in c175bcc07af7

Warning: /app/toodles.cabal was modified manually. Ignoring /app/package.yaml in favor of the cabal
         file. If you want to use the package.yaml file instead of the cabal file, then please
         delete the cabal file.
Building all executables for `toodles' once. After a successful build of all of them, only specified executables will be rebuilt.
toodles-1.0.0: configure (lib + exe)
Configuring toodles-1.0.0...
toodles-1.0.0: build (lib + exe)
Preprocessing library for toodles-1.0.0..
Building library for toodles-1.0.0..
[1 of 6] Compiling Paths_toodles    ( .stack-work/dist/x86_64-linux/Cabal-2.2.0.1/build/autogen/Paths_toodles.hs, .stack-work/dist/x86_64-linux/Cabal-2.2.0.1/build/Paths_toodles.o )
[2 of 6] Compiling Types            ( src/Types.hs, .stack-work/dist/x86_64-linux/Cabal-2.2.0.1/build/Types.o )
[3 of 6] Compiling ToodlesApi       ( src/ToodlesApi.hs, .stack-work/dist/x86_64-linux/Cabal-2.2.0.1/build/ToodlesApi.o )
[4 of 6] Compiling Parse            ( src/Parse.hs, .stack-work/dist/x86_64-linux/Cabal-2.2.0.1/build/Parse.o )
[5 of 6] Compiling Config           ( src/Config.hs, .stack-work/dist/x86_64-linux/Cabal-2.2.0.1/build/Config.o )
[6 of 6] Compiling Server           ( src/Server.hs, .stack-work/dist/x86_64-linux/Cabal-2.2.0.1/build/Server.o )
/usr/bin/ld.gold: error: /usr/lib/gcc/x86_64-linux-gnu/6/crtbeginT.o: requires dynamic R_X86_64_32 reloc against '__TMC_END__' which may overflow at runtime; recompile with -fPIC
collect2: error: ld returned 1 exit status
`gcc' failed in phase `Linker'. (Exit code: 1)

--  While building custom Setup.hs for package toodles-1.0.0 using:
      /root/.stack/setup-exe-cache/x86_64-linux/Cabal-simple_mPHDZzAJ_2.2.0.1_ghc-8.4.3 --builddir=.stack-work/dist/x86_64-linux/Cabal-2.2.0.1 build lib:toodles exe:toodles --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
The command '/bin/sh -c stack install --ghc-options '-optl-static -fPIC -optc-Os'' returned a non-zero code: 1
```

**Comment by aviaviavi** [comment]
@fbartels do you recall seeing anything like this when you were getting the docker container working?

note that i added ` -fPIC -optc-Os` from what is checked in in master right now, from seeing `https://www.fpcomplete.com/blog/2016/10/static-compilation-with-stack`. The error we get on dockerhub is the same now https://hub.docker.com/r/aviaviavi/toodles/builds/b6oq4ftbbfswrujqt8mqyzl/

**Comment by aviaviavi** [comment]
when i remove the flags completely, i get issues running the binary on alpine. i'm going to try removing the flags and using debian:stretch instead. the final image should still be on the smaller side

**Comment by aviaviavi** [comment]
fixed with a8b73a75d06683196c717f227b2dd72e80514adb

**Comment by aviaviavi** [comment]
The alpine image was still a bit smaller, so I think getting that back would be an improvement if it's an easy fix! No need to spend too much time though since the Debian route is indeed working :) 

Thanks!
